pkg/ruler/store_mock_test.go

```diff
@@ -148,6 +156,14 @@ func (m *mockRuleStore) ListRuleGroupsForUserAndNamespace(_ context.Context, use
 		return rulespb.RuleGroupList{}, nil
 	}
 
+	for i, r := range userRules {
```

Comment:
I'm not sure how `ListRuleGroupsForUserAndNamespace` got this big, but can we simplify it to just:

```
func (m *mockRuleStore) ListRuleGroupsForUserAndNamespace(_ context.Context, userID, namespace string) (rulespb.RuleGroupList, error) {
	m.mtx.Lock()
	defer m.mtx.Unlock()

	var result rulespb.RuleGroupList
	for _, r := range m.rules[userID] {
		if namespace != "" && namespace != r.Namespace {
			continue
		}

		result = append(result, &rulespb.RuleGroupDesc{
			Namespace: r.Namespace,
			Name:      r.Name,
			User:      userID,
		})
	}
	return result, nil
}
```

Reply:
Wonderful! All is work well.